### PR TITLE
proxy: plugin_proxy: Clean up allocated resources correctly

### PR DESCRIPTION
**Pull request by cosmo0920**
Signed-off-by: Hiroshi Hatake <hatake@calyptia.com>

Currently, fluent-bit frees allocated memory on every time even if a clean up callback is registered.
However, freeing allocated memory in every time is not good behavior for resource handling on Golang shared object (input plugin) side.

Also, we should keep the behavior when cleanup callback is not registered.

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [ ] Example configuration file for the change
- [ ] Debug log output from testing the change

- [ ] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.
- [ ] Run [local packaging test](./packaging/local-build-all.sh) showing all targets (including any new ones) build.
- [ ] Set `ok-package-test` label to test for all targets (requires maintainer to do).

**Documentation**

- [ ] Documentation required for this feature

**Backporting**

- [ ] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

File: src/proxy/go/go.c [hunk 1]
```diff
@@ -255,6 +255,13 @@ int proxy_go_input_cleanup(struct flb_plugin_proxy *ctx,
     if (plugin->cb_cleanup) {
         ret = plugin->cb_cleanup(allocated_data);
     }
+    else {
+        /* If cleanup callback is not registered, we need to cleanup
+         * allocated memory on fluent-bit side. */
```
Comment (pwhelan) [inline src/proxy/go/go.c:260]: Is this attempting to free a NULL pointer?

Comment (cosmo0920) [reply src/proxy/go/go.c:260]: No. Without cleanup callback, there is no freeing resource mechanism. These lines tend to free allocated resources where are allocated on input callback.

Comment (cosmo0920) [reply src/proxy/go/go.c:260]: I changed to use not NULL in https://github.com/fluent/fluent-bit/pull/6823/commits/c7c074398fb875e903cc111ecac236e971036299.

**Comment by cosmo0920** [comment]
Thank you!

**Pull request merged**